### PR TITLE
Add rjson syntax highlighter to VS Code extension

### LINKED ISSUE #805: 📎 Syntax highlighting of RJSON in VSCode extension
Now that we internally use `.rjson` files extensively, we could extend the VSCode extension to detect these files and highlight its syntax.

I would be also great if we could apply the linting too

### PR DESCRIPTION
**Pull request by antfu**
## Summary

Resolves #805.

This PR adds rjson syntax highlighting support to VS Code.

The tmGrammer is ported and modified from https://github.com/Microsoft/vscode-JSON.tmLanguage/blob/master/JSON.tmLanguage

image

### CODE REVIEW

**Comment by antfu** [comment]
Not very sure why the CI fails. Can you have a look? Thanks

File: public-packages/vscode-extension/syntaxes/RJSON.tmLanguage.json [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          "name": "comment.line.double-slash.js"
+        }
+      ]
+    },
+    "constant": {
+      "match": "\\b(?:true|false|null)\\b",
+      "name": "constant.language.json.rome"
+    },
+    "number": {
+      "match": "(?x)        # turn on extended mode\n  -?        # an optional minus\n  (?:\n    0       # a zero\n    |       # ...or...\n    [1-9]   # a 1-9 character\n    [\\d_]*     # followed by zero or more digits\n  )\n  (?:\n    (?:\n      \\.    # a period\n      \\d+   # followed by one or more digits\n    )?\n    (?:\n      [eE]  # an e character\n      [+-]? # followed by an option +/-\n      \\d+   # followed by one or more digits\n    )?      # make exponent optional\n  )?        # make decimal portion optional",
```
Comment (ematipico) [inline public-packages/vscode-extension/syntaxes/RJSON.tmLanguage.json:83]: Is this correct? It looks strange to me

**Comment by ematipico** [comment]
Have you tried to run `./rome check --apply` locally? It seems there's some issue with the tabbing

File: public-packages/vscode-extension/syntaxes/RJSON.tmLanguage.json (hunk 1 above)
Comment (antfu) [reply public-packages/vscode-extension/syntaxes/RJSON.tmLanguage.json:83]: It's copied from [VS Code's JSON stynax](https://github.com/microsoft/vscode/blob/4f00cac4346772521349ebb32f94bbba3a4e8ee7/extensions/json/syntaxes/JSONC.tmLanguage.json#L82). I just leave it as-is. 

As it's might be kind redundant, I am happy to remove the comments if necessary.

**Comment by antfu** [comment]
The CI failed with
image

Not sure if it's about formating. Is there anything I can do to fix it?

**Comment by ematipico** [comment]
There's still some problem with tabbing. I just downloaded your fork and `RJSON.tmLanguage.json` has spaces instead of tabs

**Comment by sebmck** [comment]
Love it, thank you! Sorry for taking so long to get to this PR. Just wanted to note that we are likely going to remove RJSON, but that might not be for a while, but this is still really appreciated. You can follow that discussion in #1081.

**Pull request merged**